<article class="guide">
  <carousel class="deck container-fluid">
    <slide class="row-fluid">
      <div class="col-sm-2">
        <div class="imageblock">
          <div class="content">
            <img class="img-fluid"
              src="https://raw.githubusercontent.com/COMBAT-TB/combat-tb-db/master/docs/images/logo.png"
              alt="NeoDB graph model" />
          </div>
        </div>
      </div>
      <div class="col-sm-10">
        <h3>
          <a href="https://combattb.org" target="_blank" rel="”noopener" noreferrer”>COMBAT-TB</a>
          NeoDB: fostering Tuberculosis research through integrative analysis
          using graph database technologies
        </h3>
        <br />
        <div class="paragraph">
          <p class="lead">
            Tuberculosis (TB) is a significant global health threat, with one-third of
            the population infected with its causative agent <i>Mycobacterium tuberculosis (M.tb)</i>.
            Globally, researchers have been responding with a
            plethora of heterogeneous TB databases with each focusing on different
            subsets of TB data and present limited options for data integration thus
            impeding the chances of integrative analysis. Although each database
            can provide answers to certain questions in its scope, it falls short in
            answering questions that require federated queries across multiple
            domains of biological knowledge.
          </p>
        </div>
        <div class="paragraph">
          <p class="lead">
            Under the auspices of the Computational Bacterial Analytical Tool-kit for TB
            (<a href="https://combattb.org" target="_blank" rel="”noopener" noreferrer”>COMBAT-TB</a>) project,
            we developed NeoDB, an integrated <i>M.tb</i> ‘omics’ knowledge-base. NeoDB is based on
            <a href="https://neo4j.com" target="_blank" rel="”noopener" noreferrer”>Neo4j</a>
            and enables researchers to execute complex federated queries by linking well-known, curated and
            widely used biological data resources, and supplementary TB variants data from published literature.
          </p>
        </div>
        <div class="paragraph">
          <p class="lead">
            We developed <a href="https://combattb.org/combat-tb-neodb" target="_blank" rel="”noopener"
              noreferrer”>Combat-TB-NeoDB</a>
            to enable <i>M.tb</i> researchers to execute ad hoc complex federated queries with the
            ability to explore data interactively by linking well-known, curated and widely used M.tb data resources,
            and supplementary TB variants data from published literature.
          </p>
        </div>
        <div class="paragraph">
          <p class="lead">
            <strong>The next slides contain guides on example queries that can be executed in the
              Neo4j browser. More examples can be found
              <a href="https://combattb.org/combat-tb-neodb/" target="_blank" rel="”noopener"
                noreferrer”>here</a>.</strong>
          </p>
        </div>
      </div>
    </slide>
    <slide class="row-fluid">
      <div class="col-sm-12">
        <h3>Explore Combat-TB-NeoDB</h3>
        <br />
        <div>
          <div class="col-lg-3 ng-scope">
            <h4>What is related, and how?</h4>
            <br />
            <div class="paragraph">
              <p class="lead">
                The following query shows the schema of the data. Run it by clicking the
                text box to prefill the command and then hitting
                <a tooltip-placement="left" class="circled play sl sl-play"></a>
                in the upper right.
              </p>
            </div>
            <div class="listingblock">
              <div class="content">
                <pre class="pre-scrollable code runnable">CALL db.schema.visualization</pre>
              </div>
            </div>
          </div>
          <div class="col-lg-3 ng-scope">
            <h4>Label counts</h4>
            <br />
            <div class="paragraph">
              <p class="lead">
                In Neo4j, node types are called labels. The following query
                counts the number of nodes per label.
              </p>
            </div>
            <div class="listingblock">
              <div class="content">
                <pre class="pre-scrollable code runnable">
MATCH(node) 
RETURN head(labels(node)) AS Label, 
  count(*) AS Count 
ORDER BY count DESC</pre>
              </div>
            </div>
          </div>
          <div class="col-lg-3 ng-scope">
            <h4>Relationship type counts</h4>
            <br />
            <div class="paragraph">
              <p class="lead">
                Play the following query to count the number of relationships
                per type.
              </p>
            </div>
            <div class="listingblock">
              <div class="content">
                <pre class="pre-scrollable code runnable">
MATCH()-[rel]->() 
RETURN type(rel) AS RelType, 
  count(*) AS Count 
ORDER BY count DESC</pre>
              </div>
            </div>
          </div>
          <div class="col-lg-3 ng-scope">
            <h4>What kind of nodes exist?</h4>
            <br />
            <div class="paragraph">
              <p class="lead">
                Play the following query to sample some nodes,
                reporting on property and relationship counts per node.
              </p>
            </div>
            <div class="listingblock">
              <div class="content">
                <pre class="pre-scrollable code runnable">
MATCH (n) 
WHERE rand() <= 0.1 
RETURN DISTINCT labels(n) as Labels, 
  count(*) AS SampleSize, 
  avg(size(keys(n))) as Avg_PropertyCount, 
  min(size(keys(n))) as Min_PropertyCount, 
  max(size(keys(n))) as Max_PropertyCount, 
  avg(size((n)-[]-())) as Avg_RelationshipCount, 
  min(size((n)-[]-())) as Min_RelationshipCount, 
  max(size((n)-[]-())) as Max_RelationshipCount</pre>
              </div>
            </div>
          </div>
        </div>
      </div>
    </slide>
    <slide class="row-fluid">
      <div class="col-sm-12">
        <h3>Learn how to query NeoDB using Cypher</h3>
        <p class="lead"><strong>Part 1</strong></p>
        <br />
        <div class="paragraph">
          <p class="lead">
            Combat-TB-NeoDB can be used for the analysis of TB data.
            Federated queries that are not possible in existing TB portals but possible
            in NeoDB include extracting known variants from Mtb genes, genes that
            interact with known drug targets, and prioritizing functional variants for
            further exploration. Below are some example queries.
          </p>
        </div>
        <div class="paragraph">
          <p class="lead">
            <strong>Find genes that encode proteins, limiting to results to 25</strong>
          </p>
        </div>
        <pre class="pre-scrollable code runnable">
MATCH(g:Gene)-[r:ENCODES]->(p:Protein) 
RETURN g.name as Gene, p.name as Protein 
LIMIT 25</pre>
        </figure>
        <div class="paragraph">
          <p class="lead">
            <strong>Which genes encode a protein that interacts with a known drug target</strong>
          </p>
        </div>
        <figure>
          <pre class="pre-scrollable code runnable">
MATCH p=(gene:Gene)-[:ENCODES]-(p1:Protein)-[i:INTERACTS_WITH]-(p2:Protein)-[:TARGET]-(drug:Drug)
RETURN gene.name as Gene, 
  i.score as Score, 
  p2.uniquename as Interactor, 
  drug.name as Drug
ORDER BY Score DESC</pre>
        </figure>
        <div class="paragraph">
          <p class="lead">
            <strong>Find a drugs that targets a set of proteins.</strong>
          </p>
        </div>
        <figure>
          <pre class="pre-scrollable code runnable">
MATCH(p:Protein)-[:TARGET]-(drug) WITH drug, 
  count(p) AS ProteinSetSize, collect(p.uniquename) AS ProteinSet 
WHERE ProteinSetSize > 1 
RETURN drug.name AS DrugName, 
  drug.accession AS DrugAcc, 
  ProteinSet, 
  ProteinSetSize 
ORDER BY ProteinSetSize DESC</pre>
        </figure>
        <div class="paragraph">
          <p class="lead">
            <strong>Find a drugs that are are targeted by a specific drug (Isoniazid).</strong>
          </p>
        </div>
        <figure>
          <pre class="pre-scrollable code runnable">
MATCH(drug:Drug {name: "Isoniazid"})-[r:TARGET]-(protein:Protein)
RETURN *</pre>
        </figure>
        <div class="paragraph">
          <p class="lead">
            <strong>Find proteins that are targeted by multiple drugs.</strong>
          </p>
        </div>
        <figure>
          <pre class="pre-scrollable code runnable">
MATCH(drug:Drug)-[:TARGET]-(protein) WITH protein, 
  count(drug) as DrugCount, 
  collect(drug.name) as DrugNames, 
  protein.uniquename as Proteins, 
  protein.function as ProteinFunctions 
WHERE DrugCount > 1 
RETURN Proteins, 
  ProteinFunctions, 
  DrugCount, 
  DrugNames ORDER BY DrugCount DESC</pre>
        </figure>
        <div class="paragraph">
          <p>
            For more advanced examples, see
            <a href="https://combattb.org/combat-tb-neodb/">documentation</a>.
          </p>
        </div>
      </div>
    </slide>
    <slide class="row-fluid">
      <div class="col-sm-12">
        <h3>Learn how to query NeoDB using Cypher</h3>
        <p class="lead"><strong>Part 2</strong></p>
        <br />
        <div class="paragraph">
          <p class="lead">
            Scientists at Janssen Pharmaceutica have characterized a novel resistance mechanism to Bedaquiline (BDQ),
            a second line antibiotic for the treatment of MDR-TB. The research
            <a href="https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0102135" target="_blank"
              rel="”noopener" noreferrer”>study</a>,
            published in Plos One, reported that non-target based resistance to BDQ, and cross-resistance to clofazimine
            (CFZ),
            is due to mutations in <code>Rv0678</code>, a transcriptional repressor of the genes encoding the
            MmpS5-MmpL5 efflux pump.
          </p>
        </div>
        <div>
          <div class="paragraph">
            <p>
              Here are some simple queries about <code>Rv0678</code> to help new users get acquainted with
              Cypher and NeoDB.
            </p>
          </div>
          <div class="olist arabic">
            <ol class="arabic">
              <li>
                <p>
                  <strong>Retrieve the <code>Rv0678</code> Gene node:</strong>
                </p>
                <div class="listingblock">
                  <figure>
                    <pre class="pre-scrollable code runnable">
MATCH(gene:Gene {uniquename: "Rv0678"}) RETURN Gene</pre>
                  </figure>
                </div>
                <div class="paragraph">
                  <p>Which is equivalent to:</p>
                </div>
                <div class="listingblock">
                  <figure>
                    <pre class="pre-scrollable code runnable">
MATCH(gene:Gene) WHERE gene.uniquename = "Rv0678" RETURN Gene</pre>
                  </figure>
                </div>
              </li>
              <li>
                <p>
                  <strong>
                    Find all Gene Ontology terms <code>ASSOCIATED_WITH</code><code>Rv0678</code>:
                  </strong>
                </p>
                <div class="listingblock">
                  <figure>
                    <pre class="pre-scrollable code runnable">
MATCH path=(:Gene {uniquename: "Rv0678"})-[:ENCODES]-(:Protein)-[:ASSOCIATED_WITH]-(:GOTerm)
RETURN path</pre>
                  </figure>
                </div>
              </li>

              <li>
                <p>
                  <strong>
                    Find all genes <code>ASSOCIATED_WITH</code> <code>Rv0678</code> GO terms:
                  </strong>
                </p>
                <div class="listingblock">
                  <figure>
                    <pre class="pre-scrollable code runnable">
MATCH(:Gene {uniquename: "Rv0678"})--(:Protein)--(go:GOTerm)
WITH collect(go.accession) as GoTerms 
UNWIND GoTerms as go MATCH(gene:Gene)--(:Protein)--(:GOTerm {accession: go}) 
RETURN collect(distinct(gene.name)) as Genes, 
  count(distinct(gene.name)) as GeneCount</pre>
                  </figure>
                </div>
              </li>
            </ol>
          </div>
          <div class="paragraph">
            <p>
              For more advanced examples, see
              <a href="https://combattb.org/combat-tb-neodb/">documentation</a>.
            </p>
          </div>
        </div>
      </div>
    </slide>
    <slide class="row-fluid">
      <div class="col-sm-12">
        <h3>Learn how to query NeoDB using Cypher</h3>
        <p class="lead"><strong>Part 3</strong></p>
        <br />
        <div class="paragraph">
          <p class="lead">
            Gene Ontology (GO) annotation is a way to describe the characteristics of
            gene products using a controlled vocabulary. The attributes of a gene
            product that we capture are; where in the cell it is located (Cellular Component),
            what role it has (Molecular Function) and what pathways and
            processes it is involved in (Biological Process).
          </p>
        </div>
        <div>
          <div class="paragraph">
            <p>
              Here are some simple queries about GO to help new users get acquainted with
              Cypher and NeoDB.
            </p>
          </div>
          <div class="olist arabic">
            <ol class="arabic">
              <li>
                <p>
                  <strong>
                    Find all Gene Ontology terms <code>ASSOCIATED_WITH</code><code>Rv0678</code>:
                  </strong>
                </p>
                <div class="listingblock">
                  <figure>
                    <pre class="pre-scrollable code runnable">
MATCH path=(:Gene {uniquename: "Rv0678"})-[:ENCODES]-(:Protein)-[:ASSOCIATED_WITH]-(:GOTerm)
RETURN path</pre>
                  </figure>
                </div>
              </li>

              <li>
                <p>
                  <strong>
                    Find all genes <code>ASSOCIATED_WITH</code> GO terms containing <code>methylation</code>:
                  </strong>
                </p>
                <div class="listingblock">
                  <figure>
                    <pre class="pre-scrollable code runnable">
MATCH(go:GOTerm)--()--(gene:Gene) 
WHERE tolower(go.name) CONTAINS 'methylation' 
RETURN go.name AS GoName, 
  go.accession AS GoId,
  go.ontology AS GoOntology, 
  collect(gene.name) AS Genes</pre>
                  </figure>
                </div>
              </li>
            </ol>
          </div>
          <div class="paragraph">
            <p>
              For more advanced examples, see
              <a href="https://combattb.org/combat-tb-neodb/">documentation</a>.
            </p>
          </div>
        </div>
      </div>
    </slide>
  </carousel>
</article>
